pkgs/stdenv/generic/check-meta.nix

```diff
@@ -453,4 +507,16 @@ let
 
   };
 
+  # Given a list of values which do not contain functions, return
+  # another list with duplicates removed, in undefined but
+  # deterministic order.
+  #
+  # TODO: possibly move into /lib/
+  internStringifiable = list:
```

Comment:
Nix attrsets are sorted (natively, in C++ code) -- my goal here was to take advantage of that.  Maybe I misunderstood the question though?

Reply:
If this function can easily be expressed using existing `lib` functions then this should be done, and any optimizations done here should be put into lib instead. Maybe see also #239722 and #244187

Reply:
Note: the elements of the list are not all strings.  There are patterns (see `lib.systems.inspect`).  Maybe that clarifies the reason for the `builtins.toJSON`.

Reply:
> If this function can easily be expressed using existing `lib` functions

Using `lib.unique` to remove the duplicates from a list is O(n^2).

> Maybe see also #239722

That function is O(n log n), but it merely returns a boolean indicating if all the elements are unique, rather than removing the duplicates.  More importantly, it expects a comparison function.  In this case the needed function would be a lexicographic comparison function on attrsets, which I don't believe exists in `lib`.  Are you asking me to write one?

Reply:
Isn't `<` already implemented on most everything in Nix?

Also the discussion in there talks about adding functions which return the sorted unique list instead of a boolean, this feels like what you would want.

Reply:
> Isn't `<` already implemented on most everything in Nix?

```
$ nix eval --expr '{ a=3; } < { b=2; }'
error: cannot compare a set with a set
```

Reply:
@pennae I think you told me this, did I misunderstand something?

Reply:
FWIW I think a `builtins` to do lexicographic comparison on *function-free* attrsets would be a great addition to Nix.  But it isn't there yet.

Reply:
@piegamesde I believe this is addressed by 1c68dedefcb5ac3c2b2757fc076a1f9850c4c1a1.  I would appreciate your review of that commit.